src/Extensions/LeftAndMainSubsites.php

```diff
@@ -282,14 +282,6 @@ protected function onBeforeInit()
 
         // Catch forced subsite changes that need to cause CMS reloads.
         if ($request->getVar('SubsiteID') !== null) {
-            // Clear current page when subsite changes (or is set for the first time)
-            if ($state->getSubsiteIdWasChanged()) {
-                // sessionNamespace() is protected - see for info
-                $override = $this->owner->config()->get('session_namespace');
-                $sessionNamespace = $override ? $override : get_class($this->owner);
-                $session->clear($sessionNamespace . '.currentPage');
-            }
```

Comment:
Storing the current record ID in the session was deprecated, and is being removed in https://github.com/silverstripe/silverstripe-admin/pull/1867